nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -261,14 +272,18 @@ WaypointFollower::followWaypoints()
       // Update server state
       goal_index++;
       new_goal = true;
-      if (goal_index >= goal->poses.size()) {
-        RCLCPP_INFO(
-          get_logger(), "Completed all %zu waypoints requested.",
-          goal->poses.size());
-        result->missed_waypoints = failed_ids_;
-        action_server_->succeeded_current(result);
-        failed_ids_.clear();
-        return;
+      if (goal_index == goal->poses.size()) {
+        if (current_loop_no == no_of_loops) {
+          RCLCPP_INFO(
+            get_logger(), "Completed all %zu waypoints requested.",
+            goal->poses.size());
+          result->missed_waypoints = failed_ids_;
+          action_server_->succeeded_current(result);
+          failed_ids_.clear();
+          return;
+        }
+        goal_index = 0;
```

Comment:
Log that we're starting a new loop

Reply:
Do we need to log, when we can see the status of the loop on the RViz plugin? 

I feel we can safely skip it,

Reply:
A log would be nice because this feature could be used outside of the rviz context.

Reply:
BTW, this has been added few lines below